### PR TITLE
PACT-1078 & PACT-1079

### PR DESCRIPTION
**Pull request by rwalpole**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentSummaryService.scala [hunk 1]
```diff
@@ -83,28 +88,41 @@ class ListAgentSummaryService(val stubData: StubData, val repository: AbstractRe
...
+    listAgentSummary: ListAgentSummary
+  ): Try[List[AgentSummary]] =
+    agentSummaryEntities.map { agentSummaryEntity =>
+      agentSummaryEntity.as[Option[AgentSummary]] match {
+        case Some(agentSummary) =>
+          for {
+            agentDescriptions            <- getAgentDescriptions(agentSummaryEntity, listAgentSummary)
+            agentSummaryWithDescriptions <- combineSummaryAndDescriptions(agentSummary, agentDescriptions)
+          } yield agentSummaryWithDescriptions
+        case _ => Failure(new Exception("")) // TODO(RW) what exception to use here?
```
Comment (adamretter) [inline src/main/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentSummaryService.scala:114]: Still needs to be resolved perhaps?

File: src/main/scala/uk/gov/nationalarchives/omega/api/repository/AbstractRepository.scala [hunk 2]
```diff
@@ -21,17 +21,22 @@
...
 
+import org.apache.jena.ext.xerces.util.URI
 import uk.gov.nationalarchives.omega.api.messages.reply.LegalStatus
-import uk.gov.nationalarchives.omega.api.repository.model.AgentEntity
+import uk.gov.nationalarchives.omega.api.messages.request.ListAgentSummary
+import uk.gov.nationalarchives.omega.api.repository.model.{ AgentDescriptionEntity, AgentSummaryEntity }
 
 import scala.util.Try
 
 trait AbstractRepository {
```
Comment (adamretter) [inline src/main/scala/uk/gov/nationalarchives/omega/api/repository/AbstractRepository.scala:31]: I am a bit concerned by the lack of Scaladoc. I am wondering what an "AbstractRepository" or even a "Repository" is?

File: src/main/scala/uk/gov/nationalarchives/omega/api/repository/SparqlParams.scala [hunk 3]
```diff
@@ -77,7 +77,9 @@ object SparqlParams extends AgentTypeMapper {
   private def getDateTimeMap(listAgentSummary: ListAgentSummary): Map[String, String] =
     listAgentSummary.versionTimestamp match {
       case Some("all") | Some("latest") | None => Map.empty
-      case Some(dateTimeValue) =>
+      case Some(dateTimeValue)                 =>
+        // TODO(RW) we should attempt to parse this date to ensure it is valid (see PACT-1087)
```
Comment (adamretter) [inline src/main/scala/uk/gov/nationalarchives/omega/api/repository/SparqlParams.scala:81]: With your comment in mind, would it perhaps make sense to not treat xs:dateTime as a String type internally but instead treat it as XMLGregorianCaldendar or other type?

**Review by adamretter — COMMENTED** [review]
In principle this LTGM. Just a few small questions/observations...

File: src/main/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentSummaryService.scala (hunk 1 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/business/agents/ListAgentSummaryService.scala]: Have now created a `ServiceException` class for this and other similar errors.

File: src/main/scala/uk/gov/nationalarchives/omega/api/repository/AbstractRepository.scala (hunk 2 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/repository/AbstractRepository.scala:35]: Have added Scaladoc here.

File: src/main/scala/uk/gov/nationalarchives/omega/api/repository/RepositoryUtils.scala [hunk 4]
```diff
@@ -40,17 +41,26 @@ trait RepositoryUtils extends AgentTypeMapper {
 
   def prepareParameterizedQuery(
     queryResource: String,
-    params: SparqlParams
+    params: SparqlParams,
+    extendQuery: Boolean
```
[the remark this answers is not in the document]
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/repository/RepositoryUtils.scala:57]: Have added Scaladoc here.

File: src/main/scala/uk/gov/nationalarchives/omega/api/repository/SparqlParams.scala (hunk 3 above)
Comment (rwalpole) [reply src/main/scala/uk/gov/nationalarchives/omega/api/repository/SparqlParams.scala]: I think the map can be of type `Map[String, XMLGregorianCalendar]` - and I have changed this - but the `ListAgentSummary.versionTimestamp` can also be "all" or "latest" so I can't use a date type in there.

**Pull request merged**